friends/src/App.js

```diff
@@ -0,0 +1,112 @@
+import axios from "axios";
+import React, { Component } from "react";
+import { Route, NavLink } from "react-router-dom";
+
+import FriendsList from "./components/FriendsList";
+import FriendForm from "./components/FriendForm";
+
+function Header({ name }) {
+  return (
+    <header>
+      <h1 className="main">{`${name}'s Friends List`}</h1>
+      <nav className="nav">
+        <NavLink to="/add">Add Friend</NavLink>
+        <NavLink exact to="/">
+          Friends List
+        </NavLink>
+      </nav>
+    </header>
+  );
+}
+
+class App extends Component {
+  state = {
+    config: {
+      name: "Jamelle",
+      endpoint: "http://localhost:5000/friends"
+    },
+    friends: [],
+    toUpdate: null
+  };
+
+  addFriend = (user, friend) => {
+    axios
+      .post(user, friend)
+      .then(res => this.setState({ friends: res.data }))
+      .catch(err => console.log(err));
+  };
+
+  editFriend = friend => this.setState({ toUpdate: { ...friend } });
+
+  deleteFriend = (user, id) => {
+    axios
+      .delete(`${user}/${id}`)
+      .then(res => this.setState({ friends: res.data }))
+      .catch(err => console.log(err));
+  };
+
+  getFriends = user => {
+    axios
```

Comment:
at first a little confused on the flow here, but once I got to the cdm, it was clear. Maybe put that at the top of your methods so it's clearer?